request-chroot_test.go

```diff
@@ -0,0 +1,428 @@
+// +build !plan9
+
+package sftp
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const outbound = "../outbound"
+
+func getChrootTuple(t *testing.T) (string, ClientWrapper) {
+	testRoot, err := ioutil.TempDir(os.TempDir(), "sftp")
```

Comment:
We should be using a different directory name for each test, so that we can see at a glance which test failed to cleanup if it unintentionally gets left hanging.